Problems with Emacs

Not sure if this problem is new, but I run into it for the first time today:

```
$ brew gist-logs emacs
```
Results in https://gist.github.com/b495e7725743b047172658314d37cef1.

```
$ brew install emacs --HEAD  --with-cocoa --with-ctags --with-dbus --with-glib --with-gnutls  --with-imagemagick --with-mailutils

==> Using the sandbox
==> Cloning https://github.com/emacs-mirror/emacs.git
Updating /Users/amelio/Library/Caches/Homebrew/emacs--git
==> Checking out branch master
==> ./autogen.sh
==> ./configure --disable-silent-rules --enable-locallisppath=/usr/local/share/emacs/site-lisp --infodir=/usr/local/Cellar/emacs/HEAD-4a80c8b/share/info/emacs --prefix=/usr/local/Cellar/emacs/HEAD-4a80c8b
Last 15 lines from /Users/amelio/Library/Logs/Homebrew/emacs/02.configure:
checking for MagickAutoOrientImage... yes
checking for getaddrinfo_a in -lanl... no
checking for DBUS... yes
checking for dbus_watch_get_unix_fd... yes
checking for dbus_type_is_valid... yes
checking for dbus_validate_bus_name... yes
checking for dbus_validate_path... yes
checking for dbus_validate_interface... yes
checking for dbus_validate_member... yes
checking for lgetfilecon in -lselinux... no
checking for LIBGNUTLS... yes
checking for LIBGNUTLS3... yes
checking for LIBSYSTEMD... no
configure: error: '--with-file-notification=gfile' is not supported in NextStep builds.
Consider kqueue instead.

READ THIS: https://git.io/brew-troubleshooting
If reporting this issue please do so at (not Homebrew/brew):
  https://github.com/Homebrew/homebrew-core/issues

These open issues may also help:
emacs: Add option to enable dynamic modules. https://github.com/Homebrew/homebrew-core/pull/2263
```

--- a/Formula/emacs.rb
+++ b/Formula/emacs.rb
@@ -42,6 +42,8 @@ class Emacs < Formula
   depends_on "librsvg" => :recommended
   depends_on "imagemagick" => :optional
   depends_on "mailutils" => :optional
+  # Remove this option and the --with-file-notification=gfile line below once
+  # Emacs 25 is stable (#4048)
   depends_on "glib" => :optional
 
   def install
@@ -54,7 +56,7 @@ class Emacs < Formula
       --without-x
     ]
 
-    args << "--with-file-notification=gfile" if build.with? "glib"
+    args << "--with-file-notification=gfile" if build.stable? && build.with?("glib")
 
     if build.with? "libxml2"
       args << "--with-xml2"

